elastic/eui
# [Emotion] Convert EuiCollapsibleNav and EuiCollapsibleNavGroup

## Summary

- closes https://github.com/elastic/eui/issues/6389
  - Converts `EuiCollapsibleNav` to Emotion (very little actually happening here as most of the heavy styles lifting is done by `EuiFlyout`)
  - Converts `EuiCollapsibleNavGroup` to Emotion (much more going on here - I recommend [following along by commit](https://github.com/elastic/eui/pull/6865/commits))
  - Removes `.euiCollapsibleNavGroup--[background]` and `.euiCollapsibleNavGroup--withHeading` modifier classes
- closes https://github.com/elastic/eui/issues/6776

## QA

- Confirm that [production](https://elastic.github.io/eui/#/navigation/collapsible-nav) looks the same as [staging](https://eui.elastic.co/pr_6865/#/navigation/collapsible-nav)
- `gh pr checkout 6865 && yarn storybook`
- Confirm that the `background` controls work as expected (in both light and dark mode) for `EuiCollapsibleNavGroup`

### General checklist


### Emotion checklist

See https://github.com/elastic/eui/issues/6389
